vpr/src/route/rr_graph_storage.h

```diff
@@ -0,0 +1,578 @@
+        return storage_[id].xhigh_;
+    }
+    short node_yhigh(RRNodeId id) const {
+        return storage_[id].yhigh_;
+    }
+
+    short node_capacity(RRNodeId id) const {
+        return storage_[id].capacity_;
+    }
+    short node_cost_index(RRNodeId id) const {
+        return storage_[id].cost_index_;
+    }
+
+    e_direction node_direction(RRNodeId id) const {
+        if (node_type(id) != CHANX && node_type(id) != CHANY) {
+            VPR_FATAL_ERROR(VPR_ERROR_ROUTE, "Attempted to access RR node 'direction' for non-channel type '%s'", node_type_string(id));
+        }
+        return storage_[id].dir_side_.direction;
+    }
+
+    e_side node_side(RRNodeId id) const {
+        if (node_type(id) != IPIN && node_type(id) != OPIN) {
+            VPR_FATAL_ERROR(VPR_ERROR_ROUTE, "Attempted to access RR node 'side' for non-IPIN/OPIN type '%s'", node_type_string(id));
+        }
+        return storage_[id].dir_side_.side;
+    }
+
+    /* PTC get methods */
+    short node_ptc_num(RRNodeId id) const;
+    short node_pin_num(RRNodeId id) const;   //Same as ptc_num() but checks that type() is consistent
+    short node_track_num(RRNodeId id) const; //Same as ptc_num() but checks that type() is consistent
+    short node_class_num(RRNodeId id) const; //Same as ptc_num() but checks that type() is consistent
+
+    /* PTC set methods */
+    void set_node_ptc_num(RRNodeId id, short);
+    void set_node_pin_num(RRNodeId id, short);   //Same as set_ptc_num() by checks type() is consistent
+    void set_node_track_num(RRNodeId id, short); //Same as set_ptc_num() by checks type() is consistent
+    void set_node_class_num(RRNodeId id, short); //Same as set_ptc_num() by checks type() is consistent
+
+    void set_node_type(RRNodeId id, t_rr_type new_type);
+    void set_node_coordinates(RRNodeId id, short x1, short y1, short x2, short y2);
+    void set_node_cost_index(RRNodeId, size_t new_cost_index);
+    void set_node_rc_index(RRNodeId, short new_rc_index);
+    void set_node_capacity(RRNodeId, short new_capacity);
+    void set_node_direction(RRNodeId, e_direction new_direction);
+    void set_node_side(RRNodeId, e_side new_side);
+
+    // This prefetechs hot RR node data required for optimization.
+    //
+    // Note: This is optional, but may lower time spent on memory stalls in
+    // some circumstances.
+    inline void prefetch_node(RRNodeId id) const {
+        VTR_PREFETCH(&storage_[id], 0, 0);
+    }
+
+    /****************
+     * Edge methods *
+     ****************/
+
+    // Edge initialization ordering:
+    //  1. Use reserve_edges/emplace_back_edge/alloc_and_load_edges to
+    //     initialize edges.  All edges must be added prior to calling any
+    //     methods that read edge data.
+    //
+    //     Note: Either arch_switch_inf indicies or rr_switch_inf should be
+    //     used with emplace_back_edge and alloc_and_load_edges.  Do not mix
+    //     indicies, otherwise things will be break.
+    //
```

Comment:
Note: rr_switch_inf switches are remapped versions of the arch_switch_inf switch indices that are used when we have different delays and hence different indices based on the fanout of a switch.

Reply:
Done